scripts/lisk_snapshot.sh

```diff
@@ -70,93 +81,132 @@ parse_option() {
+    bash lisk.sh stop_node -c "$SNAPSHOT_CONFIG"
+  else
     echo "X Previous snapshot is in progress, aborting."
     exit 1
   fi
 fi
 
 mkdir -p "$BACKUP_LOCATION" &> /dev/null
-echo -e "\nClearing old snapshots on disk"
+echo -e "\n$( date +'%Y-%m-%d %H:%M:%S' ) Deleting snapshots older then $DAYS_TO_KEEP day(s) in $BACKUP_LOCATION"
 find "$BACKUP_LOCATION" -name "${SOURCE_DB_NAME}*.gz" -mtime +"$DAYS_TO_KEEP" -exec rm {} \;
 
-echo -e "\nClearing old snapshot instance"
+if [ "$PGSQL_VACUUM" == "Y" ] 2> /dev/null; then
+  echo -e "\n$( date +'%Y-%m-%d %H:%M:%S' ) Executing vacuum on database '$SOURCE_DB_NAME' before copy"
+  vacuumdb --analyze --full "$SOURCE_DB_NAME" &> /dev/null
+fi
+
+echo -e "\n$( date +'%Y-%m-%d %H:%M:%S' ) Cleaning old snapshot instance, database and logs"
 bash lisk.sh stop_node -c "$SNAPSHOT_CONFIG" &> /dev/null
 dropdb --if-exists "$TARGET_DB_NAME" &> /dev/null
-createdb "$TARGET_DB_NAME" &> /dev/null
+cat /dev/null > "$LOG_LOCATION"
 
-echo -e "\nExporting active database to snapshot instance"
+echo -e "\n$( date +'%Y-%m-%d %H:%M:%S' ) Copying active database '$SOURCE_DB_NAME' to snapshot database '$TARGET_DB_NAME'"
+createdb "$TARGET_DB_NAME" &> /dev/null
 pg_dump "$SOURCE_DB_NAME" | psql "$TARGET_DB_NAME" &> /dev/null
 
-echo -e "\nClearing old log files"
-cat /dev/null > "$LOG_LOCATION"
-
-echo -e '\nBeginning snapshot verification process at '"$(date)"
+echo -e "\n$( date +'%Y-%m-%d %H:%M:%S' ) Beginning snapshot verification process"
 bash lisk.sh snapshot -s "$SNAPSHOT_ROUND" -c "$SNAPSHOT_CONFIG"
 
+MINUTES=0
 until tail -n10 "$LOG_LOCATION" | (grep -q "Snapshot finished"); do
   sleep 60
-  # TODO: Check if snapshot fails
+  
+  if [ "$( stat --format=%Y "$LOG_LOCATION" )" -le $(( $(date +%s) - ( STALL_THRESHOLD_CURRENT * 60 ) )) ]; then
+    echo -e "\n$( date +'%Y-%m-%d %H:%M:%S' ) Snapshot process is stalled for $STALL_THRESHOLD_CURRENT minutes, cleaning up and exiting"
+    bash lisk.sh stop_node -c "$SNAPSHOT_CONFIG" &> /dev/null
+    dropdb --if-exists "$TARGET_DB_NAME" &> /dev/null
+    exit 1
+  fi
+  
+  MINUTES=$(( MINUTES + 1 ))
+  if [ "$PGSQL_VACUUM" == "Y" ] 2> /dev/null; then
+    if (( MINUTES % PGSQL_VACUUM_DELAY == 0 )) 2> /dev/null; then
+      echo -e "\n$( date +'%Y-%m-%d %H:%M:%S' ) Executing vacuum on table 'mem_round' of database '$TARGET_DB_NAME'"
+      DBSIZE1=$(( $( ./pgsql/bin/psql -d "$TARGET_DB_NAME" -t -c "select pg_database_size('$TARGET_DB_NAME');" | xargs ) / 1024 / 1024 ))
+      vacuumdb --analyze --full --table 'mem_round' "$TARGET_DB_NAME" &> /dev/null
+      DBSIZE2=$(( $( ./pgsql/bin/psql -d "$TARGET_DB_NAME" -t -c "select pg_database_size('$TARGET_DB_NAME');" | xargs ) / 1024 / 1024 ))
+      echo -e "$( date +'%Y-%m-%d %H:%M:%S' ) Vacuum completed, database size: $DBSIZE1 MB => $DBSIZE2 MB"
+    fi
+  fi
 done
-echo -e '\nSnapshot verification process completed at '"$(date)"
+echo -e "\n$( date +'%Y-%m-%d %H:%M:%S' ) Snapshot verification process completed"
```

Comment:
im asking you to modify it to meet our standards, this is the purpose of a code review process.